Index: Pdu.cpp
===================================================================
--- Pdu.cpp	(revision 737)
+++ Pdu.cpp	(working copy)
@@ -70,7 +70,7 @@
 
 unsigned short Pdu::getLength() const
 {
-    return _length;
+    return this->getMarshalledSize();
 }
 
 void Pdu::setLength(unsigned short pX)
@@ -95,7 +95,7 @@
     dataStream << _pduType;
     dataStream << _protocolFamily;
     dataStream << _timestamp;
-    dataStream << _length;
+    dataStream << this->getLength();
     dataStream << _padding;
 }
 @@ -120,7 +120,7 @@ bool Pdu::operator ==(const Pdu& rhs) const
      if( ! (_pduType == rhs._pduType) ) ivarsEqual = false;
      if( ! (_protocolFamily == rhs._protocolFamily) ) ivarsEqual = false;
      if( ! (_timestamp == rhs._timestamp) ) ivarsEqual = false;
-     if( ! (_length == rhs._length) ) ivarsEqual = false;
+     if( ! (getLength() == rhs.getLength()) ) ivarsEqual = false;
      if( ! (_padding == rhs._padding) ) ivarsEqual = false;
 
     return ivarsEqual;

